src/extension.ts

```diff
@@ -21,104 +21,34 @@ nls.config({ messageFormat: nls.MessageFormat.both })();
-import { WalletTreeItem } from './explorer/wallets/WalletTreeItem';
-import { debugCommandList } from './commands/debugCommandListCommand';
-import { IFabricWallet } from './fabric/IFabricWallet';
-import { editWalletCommand } from './commands/editWalletCommand';
-import { removeWallet } from './commands/removeWalletCommand';
-import { deleteIdentity } from './commands/deleteIdentityCommand';
-import { IdentityTreeItem } from './explorer/model/IdentityTreeItem';
-import { associateWallet } from './commands/associateWalletCommand';
-import { dissociateWallet } from './commands/dissociateWalletCommand';
-import { GatewayTreeItem } from './explorer/model/GatewayTreeItem';
-import { TutorialView } from './webview/TutorialView';
-import { ContractTreeItem } from './explorer/model/ContractTreeItem';
-import { TutorialGalleryView } from './webview/TutorialGalleryView';
-import { NodeTreeItem } from './explorer/runtimeOps/connectedTree/NodeTreeItem';
+import { version as currentExtensionVersion, dependencies } from '../package.json';
 import { SettingConfigurations } from '../SettingConfigurations';
-import { exportWallet } from './commands/exportWalletCommand';
 import { UserInputUtil } from './commands/UserInputUtil';
 import { FabricWalletUtil } from './fabric/FabricWalletUtil';
 import { FabricRuntime } from './fabric/FabricRuntime';
-import { dependencies } from '../package.json';
 import { FabricRuntimeUtil } from './fabric/FabricRuntimeUtil';
-import { FabricEnvironmentRegistryEntry } from './fabric/FabricEnvironmentRegistryEntry';
-import { fabricEnvironmentConnect } from './commands/environmentConnectCommand';
-import { FabricEnvironmentManager } from './fabric/FabricEnvironmentManager';
 import { FabricGatewayHelper } from './fabric/FabricGatewayHelper';
-import { addEnvironment } from './commands/addEnvironmentCommand';
-import { deleteEnvironment } from './commands/deleteEnvironment';
-import { FabricEnvironmentTreeItem } from './explorer/runtimeOps/disconnectedTree/FabricEnvironmentTreeItem';
-import { associateIdentityWithNode } from './commands/associateIdentityWithNode';
-import { FabricNode } from './fabric/FabricNode';
-import { RuntimeTreeItem } from './explorer/runtimeOps/disconnectedTree/RuntimeTreeItem';
-
-let blockchainGatewayExplorerProvider: BlockchainGatewayExplorerProvider;
-let blockchainPackageExplorerProvider: BlockchainPackageExplorerProvider;
-let blockchainEnvironmentExplorerProvider: BlockchainEnvironmentExplorerProvider;
-let blockchainWalletExplorerProvider: BlockchainWalletExplorerProvider;
-
+import { GlobalState } from './util/GlobalState';
 export async function activate(context: vscode.ExtensionContext): Promise<void> {
 
-    const originalExtensionData: ExtensionData = context.globalState.get<ExtensionData>(EXTENSION_DATA_KEY, DEFAULT_EXTENSION_DATA);
+    const globalState: GlobalState = new GlobalState(context);
```

Comment:
I think you should make global state a singleton and move storing of the context into the global state class as each time you have to call extension util and then create a new global state